File: 프로그래머스/lv2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <cmath>
#include <algorithm>
using namespace std;

vector<int> solution(int brown, int yellow)
{
    vector<int> answer;
    int carp = brown + yellow;

    for (int i = carp / 2; i > 0; i--)
    {
        if (carp % i == 0)
        {
            int a = i;
            int b = carp / i;
            if ((a - 2) * (b - 2) == yellow)
            {
                answer.push_back(a);
                answer.push_back(b);
                break;
            }
        }
    }

    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int blocks = brown + yellow;
        
        int first = 3;

        while (first < blocks) {
            int b = first;

            if (blocks % b == 0) {
                int a = blocks / b;
                
                int upDown = a * 2;
                int side = (b - 2) * 2;
                
                if (upDown + side + yellow == blocks) {
                    return new int[]{a, b};
                }
            }
            
            first++;
        }
    
        return null;
    }
}